api/apps/v1alpha1/nemo_customizer_types.go

```diff
@@ -0,0 +1,961 @@
+)
+
+const (
+	// NemoCustomizerConditionReady indicates that the NEMO CustomizerService is ready.
+	NemoCustomizerConditionReady = "Ready"
+	// NemoCustomizerConditionFailed indicates that the NEMO CustomizerService has failed.
+	NemoCustomizerConditionFailed = "Failed"
+
+	// NemoCustomizerStatusPending indicates that NEMO CustomizerService is in pending state
+	NemoCustomizerStatusPending = "Pending"
+	// NemoCustomizerStatusNotReady indicates that NEMO CustomizerService is not ready
+	NemoCustomizerStatusNotReady = "NotReady"
+	// NemoCustomizerStatusReady indicates that NEMO CustomizerService is ready
+	NemoCustomizerStatusReady = "Ready"
+	// NemoCustomizerStatusFailed indicates that NEMO CustomizerService has failed
+	NemoCustomizerStatusFailed = "Failed"
+)
+
+// NemoCustomizerSpec defines the desired state of NemoCustomizer
+type NemoCustomizerSpec struct {
+	Image          Image                        `json:"image,omitempty"`
+	Command        []string                     `json:"command,omitempty"`
+	Args           []string                     `json:"args,omitempty"`
+	Env            []corev1.EnvVar              `json:"env,omitempty"`
+	Labels         map[string]string            `json:"labels,omitempty"`
+	Annotations    map[string]string            `json:"annotations,omitempty"`
+	NodeSelector   map[string]string            `json:"nodeSelector,omitempty"`
+	Tolerations    []corev1.Toleration          `json:"tolerations,omitempty"`
+	PodAffinity    *corev1.PodAffinity          `json:"podAffinity,omitempty"`
+	Resources      *corev1.ResourceRequirements `json:"resources,omitempty"`
+	Expose         Expose                       `json:"expose,omitempty"`
+	LivenessProbe  Probe                        `json:"livenessProbe,omitempty"`
+	ReadinessProbe Probe                        `json:"readinessProbe,omitempty"`
+	StartupProbe   Probe                        `json:"startupProbe,omitempty"`
+	Scale          Autoscaling                  `json:"scale,omitempty"`
+	Metrics        Metrics                      `json:"metrics,omitempty"`
+
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=1
+	Replicas     int    `json:"replicas,omitempty"`
+	UserID       *int64 `json:"userID,omitempty"`
+	GroupID      *int64 `json:"groupID,omitempty"`
+	RuntimeClass string `json:"runtimeClass,omitempty"`
+
+	// CustomizerConfig stores the customizer configuration
+	CustomizerConfig string `json:"customizerConfig,omitempty"`
+
+	// Scheduler Configuration
+	Scheduler Scheduler `json:"scheduler,omitempty"`
+
+	// OpenTelemetry Settings
+	OpenTelemetry OTelSpec `json:"otel,omitempty"`
+
+	// Persistent Volume Claims for storage
+	Storage StorageSpecs `json:"storage,omitempty"`
+
+	// DatabaseConfig stores the database configuration
+	DatabaseConfig *DatabaseConfig `json:"databaseConfig,omitempty"`
+
+	// WandBSecret stores the secret and encryption key for the Weights and Biases service.
+	WandBSecret *WandBSecret `json:"wandbSecret,omitempty"`
```

Comment:
I see we have `omitempty` for all these fields..
Wondering whether some of them should be required